File: clojure/anagram/src/anagram.clj
(ns anagram
  (:require [clojure.string :as str]))

(defn anagram-of? [word candidate]
  (let [lower-word (str/lower-case word)
        lower-candidate (str/lower-case candidate)]
    (and
      (not= lower-word lower-candidate)
      (= (sort lower-word) (sort lower-candidate))))) 

(defn anagrams-for [word prospect-list]
  (filter (partial anagram-of? word) prospect-list))
 

File: r/anagram/anagram.R
anagram <- function(subject, candidates) {
  sorted <- \(str) str |> tolower() |> strsplit("") |> unlist() |> sort()
  
  is_anagram <- \(candidate) {
    !identical(tolower(subject), tolower(candidate)) &&
      identical(sorted(subject), sorted(candidate))
  }

  anagrams <- candidates[sapply(candidates, is_anagram)]
  if (length(anagrams) == 0)return(c())
  
  anagrams
}


